contracts/lottery/src/lib.rs

```diff
@@ -0,0 +1,608 @@
+    pub fn create_lottery(
+        env: Env,
+        ticket_price: i128,
+        number_of_numbers: u32,
+        max_range: u32,
+        thresholds: Map<u32, u32>,
+        min_players_count: u32,
+    ) {
+        let storage = env.storage().persistent();
+        if storage
+            .get::<_, LotteryState>(&DataKey::LotteryState)
+            .is_none() {
+            panic_with_error!(&env, Error::NotInitialized);
+        }
+
+        let admin = storage.get::<_, Address>(&DataKey::Admin).unwrap();
+        admin.require_auth();
+
+        let lottery_state = storage
+            .get::<_, LotteryState>(&DataKey::LotteryState)
+            .unwrap();
+        if lottery_state == LotteryState::Active {
+            panic_with_error!(&env, Error::AlreadyActive);
+        }
+
+        if max_range < number_of_numbers {
+            panic_with_error!(&env, Error::MaxRangeTooLow);
+        }
+
+        if number_of_numbers < 2 {
+            panic_with_error!(&env, Error::NumberOfNumbersTooLow);
+        }
+
+        if thresholds.len() < 1 {
+            panic_with_error!(&env, Error::NumberOfThresholdsTooLow);
+        }
+
+        let sum_of_percentages = thresholds.values()
+            .iter()
+            .fold(0u32, |acc, percentage| acc.add(percentage));
+
+        if sum_of_percentages < 1 || sum_of_percentages > 100 {
+            panic_with_error!(&env, Error::InvalidThresholds);
+        }
+
+        for threshold_number in thresholds.keys() {
+            if threshold_number < 1 || threshold_number > number_of_numbers {
+                panic_with_error!(&env, Error::InvalidThresholds);
+            }
+        }
+
+        let lottery_number = storage
+            .get::<_, u32>(&DataKey::LotteryNumber)
+            .unwrap_or_default()
+            + 1;
+
+        storage.set(&DataKey::NumberOfNumbers, &number_of_numbers);
+        storage.set(&DataKey::MaxRange, &max_range);
+        storage.set(&DataKey::Thresholds, &thresholds);
+        storage.set(&DataKey::MinPlayersCount, &min_players_count);
+        storage.set(&DataKey::TicketPrice, &ticket_price);
+        storage.set(
+            &DataKey::Tickets,
+            &Map::<Address, Vec<LotteryTicket>>::new(&env),
+        );
+        storage.set(&DataKey::LotteryState, &LotteryState::Active);
+        storage.set(&DataKey::LotteryNumber, &lottery_number);
+
+        let topic = (Symbol::new(&env, "new_lottery_created"), lottery_number);
+        env.events().publish(topic, (number_of_numbers, max_range, thresholds, ticket_price));
+    }
+
+    /// A 'dummy' method that needs to be called by user before buying the ticket.
+    /// This a workaround to this issue https://github.com/eigerco/nebula/issues/41
+    /// 
+    /// # Arguments
+    ///
+    /// - `_` - The environment for this contract - not used here
+    /// - `by` - The address that is registering.
+    pub fn register(_: Env, by: Address) {
```

Comment:
Hmmm, this doesn't look so good. Have we asked on discord? maybe we are missing something.

Reply:
I did not. I plan to make an issue on the Soroban GH with some simple example showing what is the problem.

Reply:
Question posted on [discord](https://discord.com/channels/897514728459468821/1037066367326752818/threads/1157276028327166024). Seems it might be a bug and our workaround should be fine for now.